app/Http/Controllers/RemoteProcessingController.php

```diff
@@ -63,18 +65,20 @@ public function deleteSubmissionFile(): Response
 
         if (config('cdash.backup_timeframe') == 0) {
             // Delete the file.
-            if (Storage::delete($filename)) {
-                return response('OK', Response::HTTP_OK);
-            } else {
+            try {
+                Storage::delete($filename);
+            } catch (UnableToDeleteFile $e) {
                 return response('Deletion failed', Response::HTTP_INTERNAL_SERVER_ERROR);
             }
+            return response('OK', Response::HTTP_OK);
```

Comment:
This is now common to the two branches.  As such, you could move this outside the if-statement.